plugins/modules/host.py

```diff
@@ -256,6 +445,10 @@ def main():
         elif 'owner_group' in module.foreman_params:
             module.foreman_params['owner_type'] = 'Usergroup'
 
+        if 'interfaces_attributes' in module.foreman_params:
+          filtered = [nic for nic in ({k: v for k, v in obj.items() if v} for obj in module.foreman_params['interfaces_attributes']) if nic]
+          module.foreman_params['interfaces_attributes'] = filtered
```

Comment:
```suggestion
            filtered = [nic for nic in ({k: v for k, v in obj.items() if v} for obj in module.foreman_params['interfaces_attributes']) if nic]
            module.foreman_params['interfaces_attributes'] = filtered
```

your indentation is off ;)